patches/0041-allow-lambdas-with-source_code_hash.patch

```diff
@@ -0,0 +1,141 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Venelin <venelin@pulumi.com>
+Date: Thu, 1 Feb 2024 20:07:48 +0000
+Subject: [PATCH 41/41] allow lambdas with source_code_hash
+
+
+diff --git a/internal/service/lambda/function.go b/internal/service/lambda/function.go
+index f7a47a0df7..2250da925a 100644
+--- a/internal/service/lambda/function.go
++++ b/internal/service/lambda/function.go
+@@ -169,7 +169,7 @@ func ResourceFunction() *schema.Resource {
+ 			"filename": {
+ 				Type:         schema.TypeString,
+ 				Optional:     true,
+-				ExactlyOneOf: []string{"filename", "image_uri", "s3_bucket"},
++				ExactlyOneOf: []string{"filename", "image_uri", "s3_bucket", "source_code_hash"},
```

Comment:
Do we not need the same `ExactlyOneOf` clause on `"source_code_hash"`? If not, why not?

https://github.com/hashicorp/terraform-provider-aws/blob/97c16f28e84ab7afda6b935e19b9243510931c81/internal/service/lambda/function.go#L371-L376

Reply:
good point - I don't think it's required for the validation but it will make the error messages more consistent